packages/microservices/interfaces/microservice-configuration.interface.ts

```diff
@@ -286,7 +286,7 @@ export interface RmqOptions {
      * Type of the exchange
      * @default 'topic'
      */
-    exchangeType?: 'direct' | 'fanout' | 'topic' | 'headers';
+    exchangeType?: string;
```

Comment:
```suggestion
    exchangeType?: 'direct' | 'fanout' | 'topic' | 'headers' | string & {};
```

with this (ugly) hack we'd be able to preserve type intelisense

Reply:
@kamilmysliwiec That would also be good. pragmatic way to preserve Intellisense while keeping flexibility.

Maybe we could define a type and use it explicitly to make things even clearer.

```typescript
// packages/microservices/external/rmq-url.interface.ts:71
type AmqpMainExchangeType = 'direct' | 'fanout' | 'topic' | 'headers';

/**
 * @publicApi
 */
export type AmqpExchangeType = AmqpMainExchangeType | (string & {});
```

```typescript
// packages/microservices/interfaces/microservice-configuration.interface.ts:289
exchangeType?: AmqpExchangeType;
```

Reply:
`external` is supposed to hold files (types) that are copied from the original package, so adding a new custom type there might cause unexpected issues later on